src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/PlatformManager.java

```diff
@@ -0,0 +1,217 @@
+
+        return getPlatform(name).map(platform1 ->
+            platform1.unpublishTask(record.getPlatformData()).handle((b, throwable) -> {
+                if (b != null && throwable == null) {
+                    record.setStatus(TaskStatus.finished);
+                    tasksOps.updateTask(record);
+                }
+                return true;
+            })
+        );
+    }
+
+    /**
+     * update the given experiment on the given platform
+     * @param name The name of the platform
+     * @param experiment The experiment to update
+     * @return None if the platform was not found, false if the update failed and true if everything went fine.
+     */
+    public Optional<CompletableFuture<Boolean>> updateTask(String name, Experiment experiment) throws TaskOperationException {
+        TaskRecord record;
+
+        record = tasksOps.searchTask(name, experiment.getId()).
+                orElseThrow(() -> new TaskOperationException("Experiment is not published"));
+
+        return getPlatform(name).
+                map(platform -> platform.updateTask(record.getPlatformData(), experiment)).
+                map(stringCompletableFuture -> {
+                    CompletableFuture<Boolean> result = new CompletableFuture<Boolean>();
+                    stringCompletableFuture.whenComplete((s, throwable) -> {
+                        if (s != null && throwable == null) {
+                            record.setPlatformData(s);
+                            tasksOps.updateTask(record);
+                        }
+                        result.complete(true);
+                    });
+                    return result;
+        });
+    }
+
+    /**
+     * Parse a worker id out of the params which got passed by a platform
+     * @param name The name of the platform
+     * @param params Params passed by the platform
+     * @return A String if the platform exists
+     */
+    public Optional<String> identifyWorker(String name, Map<String, String[]> params) {
+        return getWorker(name).flatMap(worker -> worker.identifyWorker(params));
+    }
+
+    /**
+     * Get a worker record from the database which is associated with the given parameters
+     * @param name Name of the platform
+     * @param params Params which were passed by the platform to the workerservice
+     * @return A WorkerRecord if one is found
+     */
+    public Optional<WorkerRecord> getWorker(String name, Map<String, String[]> params) {
+        return identifyWorker(name, params).map(uid ->
+            workerOps.getAllWorkers().stream().
+                    filter(workerRecord1 -> workerRecord1.getIdentification().equals(uid)).findFirst().
+                    orElseGet(() -> {
+                        //create a new entry in the database
+                        WorkerRecord record = new WorkerRecord(-1, uid, name, null);
+                        record = workerOps.createWorker(record);
+                        return record;
+                    })
```

Comment:
it more readable if every stream command has its own line (also the dots are typically in the front):
i would also create a new workerOperations command to get one Worker getWorker(platform, identification)
```java
workerOps.getAllWorkers().stream()
                   .filter(workerRecord1 -> workerRecord1.getIdentification().equals(did))
                   .findFirst()
                   .orElseGet(() -> {
                        //create a new entry in the database
                        WorkerRecord record = new WorkerRecord(-1, uid, name, null);
                        record = workerOps.createWorker(record);
                        return record;
                    })